### PR TITLE
Adding cross entropy loss with softmax activation for multi class problems

### PR DESCRIPTION
**Pull request by kvmanohar22**
In Ref #39 
Named the structure `SoftmaxWithLoss` following suite of `Caffe Deep Learning Library`.

### CODE REVIEW

**Comment by Evizero** [comment]
Hi! Thanks for spending time on this.

Could you add some correctness tests (e.g. comparing input-ouput pairs to results that other implementations get), as well as some type stability tests

**Comment by kvmanohar22** [comment]
I implemented this loss in `tensorflow`, and the result comparison is as follows:

# In Julia
```julia
julia> x = 0.1:0.1:0.3
0.1:0.1:0.3
julia> y = 2
2
julia> loss = SoftmaxWithLoss()
julia> value(loss, y, x)
1.101942848229244

julia> deriv(loss, y, x)
3-element Array{Float64,1}:
  0.30061 
 -0.667775
  0.367165

julia> deriv2(loss, y, x)
3-element Array{Float64,1}:
 0.148411 
 0.0861067
 0.0      
```
# In Tensorflow
```python
loss = [1.1019429]
deriv = [array([[ 0.30060959, -0.66777503,  0.36716539]], dtype=float32)]
deriv2 =  
```

**Comment by juliohm** [comment]
I am closing this PR because we need to refactor the project to have better support for multi-class losses. The implemented interface here will not fit this generalization.
